planning/obstacle_velocity_planner/src/node.cpp

```diff
@@ -0,0 +1,689 @@
+  publishCalculationTime(calculation_time);
+  RCLCPP_INFO_EXPRESSION(
+    rclcpp::get_logger("ObstacleVelocityPlanner"), is_showing_debug_info_, "%s := %f [ms]",
+    __func__, calculation_time);
+}
+
+ObstacleVelocityPlannerData ObstacleVelocityPlannerNode::createPlannerData(
+  const Trajectory & trajectory, DebugData & debug_data)
+{
+  stop_watch_.tic(__func__);
+
+  const auto & current_pose = self_pose_listener_.getCurrentPose()->pose;
+  const double current_vel = current_twist_ptr_->twist.linear.x;
+  const double current_accel = calcCurrentAccel();
+
+  // create planner_data
+  ObstacleVelocityPlannerData planner_data;
+  planner_data.current_time = now();
+  planner_data.traj = trajectory;
+  planner_data.current_pose = current_pose;
+  planner_data.current_vel = current_vel;
+  planner_data.current_acc = current_accel;
+  planner_data.target_obstacles =
+    filterObstacles(*in_objects_ptr_, trajectory, current_pose, current_vel, debug_data);
+
+  // print calculation time
+  const double calculation_time = stop_watch_.toc(__func__);
+  RCLCPP_INFO_EXPRESSION(
+    rclcpp::get_logger("ObstacleVelocityPlanner"), is_showing_debug_info_, "  %s := %f [ms]",
+    __func__, calculation_time);
+
+  return planner_data;
+}
+
+double ObstacleVelocityPlannerNode::calcCurrentAccel() const
+{
+  const double diff_vel = current_twist_ptr_->twist.linear.x - prev_twist_ptr_->twist.linear.x;
+  const double diff_time = std::max(
+    (rclcpp::Time(current_twist_ptr_->header.stamp) - rclcpp::Time(prev_twist_ptr_->header.stamp))
+      .seconds(),
+    1e-03);
+
+  const double accel = diff_vel / diff_time;
+
+  return lpf_acc_ptr_->filter(accel);
+}
+
+std::vector<TargetObstacle> ObstacleVelocityPlannerNode::filterObstacles(
+  const PredictedObjects & predicted_objects, const Trajectory & traj,
+  const geometry_msgs::msg::Pose & current_pose, const double current_vel, DebugData & debug_data)
+{
+  const auto time_stamp = rclcpp::Time(predicted_objects.header.stamp);
+
+  // calcualte decimated trajectory
+  const auto trimmed_traj = trimTrajectoryFrom(traj, current_pose.position);
+  const auto decimated_traj =
+    decimateTrajectory(trimmed_traj, obstacle_filtering_param_.decimate_trajectory_step_length);
+  if (decimated_traj.points.size() < 2) {
+    return {};
+  }
+
+  // calcualte decimated trajectory polygons
+  const auto decimated_traj_polygons = polygon_utils::createOneStepPolygons(
+    decimated_traj, vehicle_info_, obstacle_filtering_param_.detection_area_expand_width);
+  debug_data.detection_polygons = decimated_traj_polygons;
+
+  std::vector<TargetObstacle> target_obstacles;
+  for (const auto & predicted_object : predicted_objects.objects) {
+    // filter object whose label is not cruised or stopped
+    const bool is_target_obstacle =
+      planner_ptr_->isStopObstacle(predicted_object.classification.front().label) ||
+      planner_ptr_->isCruiseObstacle(predicted_object.classification.front().label);
```

Comment:
I think these function name is better to be isStopObstaleType / isCruseObstacleType

Reply:
Thanks for your comment.

I feel that this function means checking if it "is a stop obstacle" or not. Checking if it is a stop obstacle type sounds weird.
So isStopObstacle makes sense to me.

If you have a strong opinion, I'll change my mind!

Reply:
OK! Please keep the original name.